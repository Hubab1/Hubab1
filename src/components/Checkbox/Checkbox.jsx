import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from '@material-ui/core/Checkbox';
import { css } from 'emotion';

const checkboxRoot = css`
    padding-top: 0 !important;
    padding-left: 0 !important;
    padding-right: 8px !important;
`;
const checkboxText = css`
    font-size: 12px;
    text-align: left;
`;

const gridContainer = css`
    padding: 20px 0 20px 0;
    text-align: left;
    display: flex;
    align-items: flex-start;
`;

export default function LabeledCheckbox({ name, value, checked, onChange, error, label }) {
    return (
        <div className={gridContainer}>
            <Checkbox
                name={name}
                onChange={onChange}
                checked={checked}
                value={value}
                error={error}
                classes={{ root: checkboxRoot }}
            />
            <span className={checkboxText}>{label}</span>
        </div>
    );
}

LabeledCheckbox.propTypes = {
    name: PropTypes.string,
    onChange: PropTypes.func,
    checked: PropTypes.bool,
    value: PropTypes.any,
    error: PropTypes.any,
    checkboxRoot: PropTypes.any,
    label: PropTypes.string,
};
