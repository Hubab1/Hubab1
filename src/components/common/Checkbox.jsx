import React from 'react';
import Checkbox from '@material-ui/core/Checkbox';
import Grid from '@material-ui/core/Grid';
import { css } from 'emotion';

const noPaddingTop = css`
    padding-top: 0 !important;
`
const checkboxText = css`
    font-size: 12px;
    text-align: left;
`

const textLeft = css`
    text-align: left;
`

const gridContainer = css`
    padding: 20px 0 20px 0;
`

export default function ({name, value, checked, onChange, error, label}) {
    return (
        <div className={gridContainer}>
            <Grid container spacing={0} alignItems="flex-start">
                <Grid item xs={2}>
                    <Checkbox
                        name={name}
                        onChange={onChange}
                        checked={checked}
                        value={value}
                        error={error}
                        classes={{ root: noPaddingTop }}
                    />
                </Grid>
                <Grid classes={{root: textLeft}} item xs={10}>
                    <span className={checkboxText}>{label}</span>
                </Grid>
            </Grid>
        </div>
    )
}
