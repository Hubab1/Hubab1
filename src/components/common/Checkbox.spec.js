import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from './Checkbox';

let defaultProps;

beforeEach(() => {
    defaultProps = {
        name: 'checky',
        checked: false,
        onChange: ()=>{},
        error: false,
        label: 'I authorize Nestio to obtain, on behalf of Landlord'
    }
})

it('Matches snapshot', function() {
    expect(
        renderer.create(<Checkbox {...defaultProps}/>)
    ).toMatchSnapshot();
})
