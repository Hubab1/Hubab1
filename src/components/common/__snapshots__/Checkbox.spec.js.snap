// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="css-17t7hmy"
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root css-14bneh MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
    error={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        checked={false}
        className="PrivateSwitchBase-input-4"
        data-indeterminate={false}
        disabled={false}
        name="checky"
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
  </span>
  <span
    className="css-15fyth5"
  >
    I authorize Nestio to obtain, on behalf of Landlord
  </span>
</div>
`;
